tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/PublicAssetMethodContract.scala

```diff
@@ -0,0 +1,654 @@
+           |    [
+           |      "PublicAsset/set", {
+           |        "accountId": "$ACCOUNT_ID",
+           |        "create": {
+           |          "4f29": {
+           |            "blobId": "${uploadResponse.blobId}"
+           |          }
+           |        }
+           |      }, "0"
+           |    ]
+           |  ]
+           |}""".stripMargin)
+    .when()
+      .post()
+    .`then`
+      .statusCode(HttpStatus.SC_OK)
+      .contentType(JSON)
+      .extract()
+      .body()
+      .asString()
+
+    // Then the request should return notCreated
+    assertThatJson(response)
+      .inPath("methodResponses[0]")
+      .isEqualTo(
+        s"""[
+           |    "error",
+           |    {
+           |        "type": "unknownMethod",
+           |        "description": "Missing capability(ies): com:linagora:params:jmap:public:assets"
+           |    },
+           |    "0"
+           |]""".stripMargin)
+  }
+
+  @Test
+  def createShouldFailWhenInvalidPropertyInCreationRequest(): Unit = {
+    val response: String = `given`()
+      .body(
+        s"""{
+           |  "using": ["urn:ietf:params:jmap:core", "com:linagora:params:jmap:public:assets"],
+           |  "methodCalls": [
+           |    [
+           |      "PublicAsset/set", {
+           |        "accountId": "$ACCOUNT_ID",
+           |        "create": {
+           |          "4f29": {
+           |            "invalid1": "invalid-blob-id"
+           |          }
+           |        }
+           |      }, "0"
+           |    ]
+           |  ]
+           |}""".stripMargin)
+    .when()
+      .post()
+    .`then`
+      .statusCode(HttpStatus.SC_OK)
+      .contentType(JSON)
+      .extract()
+      .body()
+      .asString()
+
+    // Then the request should return notCreated
+    assertThatJson(response)
+      .inPath("methodResponses[0][1].notCreated")
+      .isEqualTo(
+        s"""{
+           |    "4f29": {
+           |        "type": "invalidArguments",
+           |        "description": "Some unknown properties were specified",
+           |        "properties": [
+           |            "invalid1"
+           |        ]
+           |    }
+           |}""".stripMargin)
+  }
+
+  @Test
+  def creationSetShouldSupportBackReferences(): Unit = {
+    val uploadResponse: UploadResponse = uploadAsset()
+
+    val request: String =
+      s"""{
+         |  "using": ["urn:ietf:params:jmap:core", "com:linagora:params:jmap:public:assets"],
+         |  "methodCalls": [
+         |    [
+         |      "PublicAsset/set", {
+         |        "accountId": "$ACCOUNT_ID",
+         |        "create": {
+         |          "clientId1": {
+         |            "blobId": "${uploadResponse.blobId}"
+         |          }
+         |        }
+         |      }, "c1"
+         |    ],
+         |    [
+         |			"Core/echo",
+         |			{ "arg1": "#clientId1" },
+         |			"c2"
+         |		]
+         |  ]
```

Comment:
weird indent

Reply:
![image](https://github.com/linagora/tmail-backend/assets/81145350/e1921636-b66a-40ec-b108-5467a7b94616)
I don't know
I tried to re-format it again, but it was the same

Reply:
You can skip it. No problem to me.

Reply:
you can use this one https://www.browserling.com/tools/tabs-to-spaces